app/controllers/github_callback_controller.rb

```diff
@@ -0,0 +1,31 @@
+class GithubCallbackController < ApplicationController
+  # Disable CSRF verification for the endpoint GitHub uses
+  skip_before_action :verify_authenticity_token, only: [:receive]
+
+  def receive
+    # Ignore all but `push` events
+    if request.headers["X-GitHub-Event"] != "push"
```

Comment:
Ruby style is to prefer unless here: `unless request.headers["X-GitHub-Event"] == “push"`